packages/api/lambdas/bulk-operation.js

```diff
@@ -106,35 +72,37 @@ async function bulkGranuleDelete(
   const deletedGranules = [];
   const forceRemoveFromCmr = payload.forceRemoveFromCmr === true;
   const granuleIds = await getGranuleIdsForPayload(payload);
-  const granuleModel = new GranuleModel();
-  const knex = await getKnexClient({ env: process.env });
+  const knex = await getKnexClient();
 
   await pMap(
     granuleIds,
     async (granuleId) => {
-      let dynamoGranule;
       let pgGranule;
-
-      try {
-        dynamoGranule = await granuleModel.getRecord({ granuleId });
-      } catch (error) {
-        if (error instanceof RecordDoesNotExist) {
-          log.info(`Granule ${granuleId} does not exist or was already deleted, continuing`);
-          return;
-        }
-        throw error;
+      let dynamoGranule;
+      const granulePgModel = new GranulePgModel();
+      const dynamoGranuleModel = new GranuleModel();
+      const pgGranuleRecords = await granulePgModel.search(knex, {
+        granule_id: granuleId,
+      });
+      if (pgGranuleRecords.length > 1) {
+        log.warn(`Granule ID ${granuleId} is not unique across collections, cannot delete granule`);
+        return;
       }
-
-      if (dynamoGranule && dynamoGranule.published && forceRemoveFromCmr) {
-        ({ pgGranule, dynamoGranule } = await unpublishGranuleFunc(knex, dynamoGranule));
+      if (pgGranuleRecords.length === 0) {
+        log.info(`Granule ${granuleId} does not exist or was already deleted, continuing`);
+        return;
+      }
+      pgGranule = pgGranuleRecords[0];
```

Comment:
This should be refactored to use getUniquePgGranuleByGranuleId

Reply:
Ah yeah.

https://github.com/nasa/cumulus/pull/2415/commits/4f1b51bf2f50c5f55979354927cba8b89e71404a

Also changed the error type we're throwing to be a bit more explicit and so we could check it in this function.